driver/vm.go

```diff
@@ -337,8 +339,22 @@ func (vm *VirtualMachine) PowerOn() error {
 	return err
 }
 
-func (vm *VirtualMachine) WaitForIP(ctx context.Context) (string, error) {
-	return vm.vm.WaitForIP(ctx)
+func (vm *VirtualMachine) WaitForIP(ctx context.Context) (ip string, err error) {
```

Comment:
The new code is roughly based on how `vm.vm.WaitForIP(ctx)` works, by watching for changes to `guest.ipAddress`.  Unlike the original function, though, this imposes the additional restrictions that the returned IP address must:
* be valid
* not be [link-local unicast](https://en.wikipedia.org/wiki/Link-local_address#IPv6)